pallets/utxo/src/tests.rs

```diff
@@ -740,3 +681,1138 @@ proptest! {
+//         assert!(UtxoStore::<Test>::contains_key(new_utxo_hash));
+//         let new_utxo = tx.outputs[1].clone();
+//
+//         if let OutputData::NftMintV1 {
+//             ref mut token_id, ..
+//         } = nft_data
+//         {
+//             *token_id = TokenId::new_asset(H256::random());
+//         }
+//         let tx = Transaction {
+//             inputs: vec![TransactionInput::new_empty(new_utxo_hash.clone())],
+//             outputs: vec![TransactionOutput {
+//                 value: 0,
+//                 destination: Destination::Pubkey(alice_pub_key),
+//                 data: Some(nft_data.clone()),
+//             }],
+//             time_lock: Default::default(),
+//         }
+//         .sign_unchecked(&[new_utxo], 0, &alice_pub_key);
+//         // Submit
+//         assert!(UtxoStore::<Test>::contains_key(H256::from(new_utxo_hash)));
+//         frame_support::assert_err_ignore_postinfo!(
+//             Utxo::spend(Origin::signed(H256::zero()), tx),
+//             "digital data has already been minted"
+//         );
+//     });
+// }
+
+// This macro using for the fast creation and sending a tx
+macro_rules! test_tx {
+    ($data: ident, $checking: tt, $err: expr) => {
+        execute_with_alice(|alice_pub_key| {
+            let (utxo0, input0) = tx_input_gen_no_signature();
+            let output_new = TransactionOutput {
+                value: 1,
+                destination: Destination::Pubkey(alice_pub_key),
+                data: Some($data.clone()),
+            };
+            let tx = Transaction {
+                inputs: vec![input0],
+                outputs: vec![output_new],
+                time_lock: Default::default(),
+            }
+            .sign_unchecked(&[utxo0], 0, &alice_pub_key);
+            let new_utxo_hash = tx.outpoint(0);
+            let (_, init_utxo) = genesis_utxo();
+            // Send
+            assert!(UtxoStore::<Test>::contains_key(H256::from(init_utxo)));
+            // We can check what error we are expecting
+            if stringify!($checking) == "Err" {
+                frame_support::assert_err_ignore_postinfo!(
+                    Utxo::spend(Origin::signed(H256::zero()), tx),
+                    $err
+                );
+                assert!(UtxoStore::<Test>::contains_key(H256::from(init_utxo)));
+                assert!(!UtxoStore::<Test>::contains_key(new_utxo_hash));
+            } else if stringify!($checking) == "Ok" {
+                // We can check is that success
+                assert_ok!(Utxo::spend(Origin::signed(H256::zero()), tx));
+                assert!(!UtxoStore::<Test>::contains_key(H256::from(init_utxo)));
+                assert!(UtxoStore::<Test>::contains_key(new_utxo_hash));
+            }
+        });
+    };
+}
+
+#[test]
+fn test_tokens_issuance_empty_ticker() {
+    // Ticker empty
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: vec![],
+        amount_to_issue: 0,
```

Comment:
The same issue I talked about IS NOT fixed. Let me explain:

If you're testing that token_ticker is too long or empty, then every other property should be valid. The only reason this test is passing is because of the order of how you're doing your checks. That's wrong.

Meaning:

Amount to issue should be non-zero in this test 
number of decimals should be non-zero in this test
metadata_uri should be a valid value in this test

Meaning:

If you put a valid token ticker, the test should produce a valid token.

The order of validation that you do in your implementation should not affect the result of your test. I should be able to go to your validation function and change the order of token_ticker and number_of_decimals checks, and your test should still pass. This is now not the case.

Please fix this for this and all other tests. The tests now in their current format are a disaster for updating the validation function.

Reply:
Okay, I'm check them all

Reply:
Fixed all what I found